proto/scene-format.proto

```diff
@@ -0,0 +1,62 @@
+syntax = "proto3";
+
+package turbocow;
+
+message Scene {
+	enum MaterialType {
+		SOLID = 0;
+		GLASS = 1;
+		// other matherials
+	}
+	message Vector3 {
+		float x = 2;
+		float y = 3;
+		float z = 4;
+	}
+	message Transform {
+		int32 id = 5;
+		Vector3 translation = 6;
+		Vector3 rotation = 7;
+		Vector3 scale = 8;
+		int32 parent_id = 9;
```

Comment:
All objects should have a transform so object id is more than enought

Reply:
@Brookoko not all object may have transform, so it would be better to use a map both for all transformation matrices and for mesh Objecta if we have such an opportunity.

Reply:
Which object doesn't have transform? Every light has transform, every camera has transform and every object that will be rendered has transform. In 3d scene, there is no object without transform (let's forget about skybox okay)

Reply:
> let's forget about skybox okay

By the way, we should not forget to add it once we iron out other things. It seems that we will need to support skybox and _skysphere_ (based on today's lecture)

Any ideas about that?

Reply:
> Which object doesn't have transform? Every light has transform, every camera has transform and every object that will be rendered has transform. In 3d scene, there is no object without transform (let's forget about skybox okay)

When saying "no transform" I ment the default transform matrix. I mean, if all of the 20 objects for example do have specific coordinates and sizes and define a scene by themselves, it would be strangle to write 20 default matrixes in our format, wouldn't it? But from the other side it would play it safe if we define them..

Reply:
You will not write matrices. You will be specify position, rotation and scale. And matrix generation is up to user. And if you want to use default value, you can just skip a parameter.

Reply:
Regarding skybox, it is usually described as separate entity, not as a light. But it has the same properties such as intensity and color. And also it can have material